### PR TITLE
Ios delta pause bugfix like on android

### LINKED ISSUE #6093: Large delta after pause/resume (iOS)
Similar to #1196

#### Issue details
If a game is paused for a long time, after resuming, Gdx.graphics.getDeltaTime() will give the delta of the entire pause duration. This causes freeze/stutter on resume as the game tries to process possibly hours worth of delta. This was fixed for the android version but not for iOS which makes it inconsistent across backends.

#### Version of LibGDX and/or relevant dependencies
_Please provide the version(s) affected._
1.9.10

#### Please select the affected platforms
- [ ] Android
- [x] iOS (robovm)
- [ ] iOS (MOE)
- [ ] HTML/GWT
- [ ] Windows
- [ ] Linux
- [ ] MacOS

### PR DESCRIPTION
**Pull request by Armaxxx**
Should fix #6093

### CODE REVIEW

**Comment by MobiDevelop** [comment]
Please use separate branches for separate PRs.

**Comment by Armaxxx** [comment]
Oh you're completely right, how silly of me, thought I'd branch them before the fact. Will redo the PR.

**Comment by Armaxxx** [comment]
There, sorry for the revert, now this branch only holds the fix.

File: backends/gdx-backend-robovm/src/com/badlogic/gdx/backends/iosrobovm/IOSGraphics.java [hunk 1]
```diff
@@ -59,7 +59,7 @@
 public class IOSGraphics extends NSObject implements Graphics, GLKViewDelegate, GLKViewControllerDelegate {
 
 	private static final String tag = "IOSGraphics";
-
+	volatile boolean resume = false;
```
Comment (obigu) [inline backends/gdx-backend-robovm/src/com/badlogic/gdx/backends/iosrobovm/IOSGraphics.java:62]: We can move it with the rest of variable declarations.

I wonder if we need it `volatile`. I'm not sure `resume()` or `draw()` can be called from any other than the main thread. I guess it's not a bad idea to be on the safe side do the same as we do with `paused`.

File: backends/gdx-backend-robovm/src/com/badlogic/gdx/backends/iosrobovm/IOSGraphics.java [hunk 2]
```diff
@@ -308,6 +308,7 @@ public void resume () {
 			for (LifecycleListener listener : listeners) {
 				listener.resume();
 			}
+			resume = true;
```
Comment (obigu) [inline backends/gdx-backend-robovm/src/com/badlogic/gdx/backends/iosrobovm/IOSGraphics.java:311]: Can we move it outside the synchronized block?

File: backends/gdx-backend-robovm/src/com/badlogic/gdx/backends/iosrobovm/IOSGraphics.java [hunk 3]
```diff
@@ -354,9 +355,18 @@ public void draw (GLKView view, CGRect rect) {
...
+		if (!resume) {
+			deltaTime = (time - lastFrameTime) / 1000000000.0f;
+		} else {
+			deltaTime = 0;
+		}
 		lastFrameTime = time;
 
+		if (resume) {
+			resume = false;
+		}
```
Comment (obigu) [inline backends/gdx-backend-robovm/src/com/badlogic/gdx/backends/iosrobovm/IOSGraphics.java:368]: Minor thing but we can probably remove this block and set resume to false on the previous conditional that checks resume value.

**Review by obigu — CHANGES_REQUESTED** [review]
Thanks for the PR. Still have no tested it, just some style stuff.

Regarding the code format changes, we usually try to only apply style changes to new code to keep git history and PRs easier to read.

File: backends/gdx-backend-robovm/src/com/badlogic/gdx/backends/iosrobovm/IOSGraphics.java [hunk 4]
```diff
@@ -354,9 +355,18 @@ public void draw (GLKView view, CGRect rect) {
 		}
 
 		long time = System.nanoTime();
-		deltaTime = (time - lastFrameTime) / 1000000000.0f;
+		// After pause deltaTime can have somewhat huge value that destabilizes the mean, so let's cut it off
```
Comment (obigu) [inline backends/gdx-backend-robovm/src/com/badlogic/gdx/backends/iosrobovm/IOSGraphics.java:358]: Remove " that destabilizes the mean", this had sense on Android when we applied a mean on `delta`

Comment (Armaxxx) [reply backends/gdx-backend-robovm/src/com/badlogic/gdx/backends/iosrobovm/IOSGraphics.java]: Good as done.

File: backends/gdx-backend-robovm/src/com/badlogic/gdx/backends/iosrobovm/IOSGraphics.java (hunk 1 above)
Comment (Armaxxx) [reply backends/gdx-backend-robovm/src/com/badlogic/gdx/backends/iosrobovm/IOSGraphics.java]: It's been moved but it was kept volatile

File: backends/gdx-backend-robovm/src/com/badlogic/gdx/backends/iosrobovm/IOSGraphics.java (hunk 2 above)
Comment (Armaxxx) [reply backends/gdx-backend-robovm/src/com/badlogic/gdx/backends/iosrobovm/IOSGraphics.java]: Good as done

File: backends/gdx-backend-robovm/src/com/badlogic/gdx/backends/iosrobovm/IOSGraphics.java (hunk 3 above)
Comment (Armaxxx) [reply backends/gdx-backend-robovm/src/com/badlogic/gdx/backends/iosrobovm/IOSGraphics.java]: You're right, it's kind of redundant.

Comment (obigu) [reply backends/gdx-backend-robovm/src/com/badlogic/gdx/backends/iosrobovm/IOSGraphics.java]: thanks for the changes, I'll test it as soon as I can

**Review by obigu — APPROVED** [review]

**Comment by obigu** [comment]
Tested, seems to we working, thanks!

**Review by MrStahlfelge — CHANGES_REQUESTED** [review]
There are a lot of unneccesary whitespace changes, please reduce the changes to the necessary ones

**Pull request merged**